source/code/plugins/out_oms_blob.rb

```diff
@@ -222,24 +233,50 @@ def commit_blocks(uri, blocks_committed, blocks_uncommitted, file_path)
 
       blocklist_uri = URI.parse("#{uri.to_s}&comp=blocklist")
       request_id = SecureRandom.uuid
-      put_blocklist_req = create_blob_put_request(blocklist_uri, commit_msg, request_id, file_path)
+      put_blocklist_req = create_blob_put_request(blocklist_uri, commit_msg, request_id, file_path, get_etag = true)
       http = OMS::Common.create_secure_http(blocklist_uri, @proxy_config)
-      OMS::Common.start_request(put_blocklist_req, http)
+      response = OMS::Common.start_request(put_blocklist_req, http, ignore404 = false, return_entire_response = true)
+
+      begin
+        etag_quoted = response.to_hash["etag"]
+        if etag_quoted.is_a?(Array)
+          etag_quoted = etag_quoted[0]
+        end
+        etag = etag_quoted.gsub(/"/, "")
+      rescue => e
+        @log.error("Cannot extract ETag from BLOB response #{response}. Error:'#{e}'")
+        etag = ""
+      end
+      return etag
     end # commit_blocks
 
-    def notify_blob_upload_complete(uri, data_type, custom_data_type)
+    # Notify ODS that we have completed uploading to the BLOB
+    # Parameters:
+    #   uri: URI. blob URI
+    #   data_type: string. DataTypeId of the data
+    #   custom_data_type: string. CustomDataType of the CustomLog
+    #   offset_blob_size: int. Amount of data that BLOB contained before we appended to it
+    #   sent_size: int. Amount of data we appended to the BLOB
+    #   etag: string. ETag from the BLOB for this data
+    def notify_blob_upload_complete(uri, data_type, custom_data_type, offset_blob_size, sent_size, etag)
       data_type_id = data_type
       if !custom_data_type.nil?
         data_type_id = "#{data_type}.#{custom_data_type}"
       end
 
+      # Remove SAS token from the URL
+      blob_url = uri.to_s.split("?")[0]
```

Comment:
Have found a solution here: https://stackoverflow.com/questions/10410523/removing-a-part-of-a-url-with-ruby

Will change to use `uri.fragment = uri.query = nil`